src/state/user/hooks.tsx

```diff
@@ -495,3 +496,25 @@ export function useToggleTopTrendingTokens(): () => void {
   const dispatch = useDispatch<AppDispatch>()
   return useCallback(() => dispatch(toggleTopTrendingTokens()), [dispatch])
 }
+
+export const useUserFavoriteTokens = (chainId: ChainId | undefined) => {
+  const dispatch = useDispatch<AppDispatch>()
+  const favoriteTokens = useSelector((state: AppState) => {
+    if (!chainId) {
+      return undefined
+    }
+
+    if (!state.user.favoriteTokensByChainId) {
+      return undefined
+    }
```

Comment:
should we merge 2 conditions?
if (!chainId || !state.user.favoriteTokensByChainId) return undefined

Reply:
or merge them with below `return` to oneline: 
```
    return (chainId && state.user.favoriteTokensByChainId?.[chainId]) ?? undefined
```

Reply:
Too fast bro. I haven't read this